### PR TITLE
xonsh arguments improvement

### PR DESCRIPTION
**Pull request by anki-code**
Hello! xonsh is the most interesting thing in world ocean! Thanks!

In this PR xonsh arguments improved with love:

1. Capitalized first letter in args help text.
2. Used [argparse "version" action](https://docs.python.org/3.5/library/argparse.html#action) to show version.
3. Suppressed old deprecated `--config-path` (deprecated since 2018).
4. ~Disabled the welcome screen if `--no-rc` was passed.~ (reverted)
5. Improved formatting of help for `--shell-type`:

    Before - ugly "usage" lines and description:
    ```
    $ xonsh --help
    usage: xonsh [-h] [-V] [-c COMMAND] [-i] [-l] [--rc RC [RC ...]] [--no-rc] [--no-script-cache]
                [--cache-everything] [-D ITEM]
                [--shell-type {b,best,d,dumb,ptk,ptk1,ptk2,prompt-toolkit,prompt_toolkit,prompt-toolkit1,prompt-toolkit2,prompt-toolkit3,prompt_toolkit3,ptk3,rand,random,rl,readline}]
                [--timings]
    ...
    -D ITEM               Define an environment variable, in the form of -DNAME=VAL. May be used many times.
    --shell-type {b,best,d,dumb,ptk,ptk1,ptk2,prompt-toolkit,prompt_toolkit,prompt-toolkit1,prompt-toolkit2,prompt-toolkit3,prompt_toolkit3,ptk3,rand,random,rl,readline}
                            What kind of shell should be used. Possible options: readline, prompt_toolkit, random.
                            Warning! If set this overrides $SHELL_TYPE variable.
    ...
    ```

    After - cute usage line and good formatted description:
    ```
    $ xonsh --help
    usage: xonsh [-h] [-V] [-c COMMAND] [-i] [-l] [--rc RC [RC ...]] [--no-rc] [--no-script-cache]
                [--cache-everything] [-D ITEM] [--shell-type SHELL_TYPE] [--timings]
                [script-file] 
    ...
    -D ITEM               Define an environment variable, in the form of -DNAME=VAL. May be used many times.
    --shell-type SHELL_TYPE
                            What kind of shell should be used. Possible options: b, best, d, dumb, ptk, ptk1, ptk2,
                            prompt-toolkit, prompt_toolkit, prompt-toolkit1, prompt-toolkit2, prompt-toolkit3,
                            prompt_toolkit3, ptk3, rand, random, rl, readline. Warning! If set this overrides
                            $SHELL_TYPE variable.
    ...
    ```

### CODE REVIEW

**Comment by anki-code** [comment]
Ready to merge.

File: xonsh/main.py [hunk 1]
```diff
@@ -128,43 +128,41 @@ def parser():
...
     p.add_argument(
         "-V",
         "--version",
-        dest="version",
-        action="store_true",
-        default=False,
-        help="show version information and exit",
+        action="version",
+        help="Show version information and exit.",
+        version="/".join(("xonsh", __version__)),
```
Comment (scopatz) [inline xonsh/main.py:138]: I think f"xonsh/{__version__}" would be better here as the value

File: xonsh/main.py [hunk 2]
```diff
@@ -446,8 +442,10 @@ def func_sig_ttin_ttou(n, f):
             # enter the shell
             env["XONSH_INTERACTIVE"] = True
             ignore_sigtstp()
-            if env["XONSH_INTERACTIVE"] and not any(
-                os.path.isfile(i) for i in env["XONSHRC"]
+            if (
+                env["XONSH_INTERACTIVE"]
+                and not args.norc
```
Comment (scopatz) [inline xonsh/main.py:447]: I think that the welcome screen should print `--no-rc` was passed in.  With this option xonsh, should function as if you were in a clean environment.

Comment (scopatz) [reply xonsh/main.py:447]: That is how I have thought about `--no-rc` but am open to other opinions.  Thoughts @xonsh/xore?

**Comment by scopatz** [comment]
Thanks for putting this @anki-code! Just a couple of thoughts

File: xonsh/main.py (hunk 1 above)
Comment (anki-code) [reply xonsh/main.py]: Yeah, it looks better! Changed.

File: xonsh/main.py (hunk 2 above)
Comment (anki-code) [reply xonsh/main.py]: I understand the logic and revert this change. Please no discuss in this micro changes...

**Comment by anki-code** [comment]
Ready to merge.

**Comment by scopatz** [comment]
Thanks @anki-code!

**Pull request merged**